financial_assistant/src/retrieval.py

```diff
@@ -118,9 +119,13 @@ def get_vectorstore_retriever(
     embedding_model = load_embedding_model(embedding_model_info)
 
     # Instantiate the vectorstore with an explicit in-memory configuration
-    vectorstore = Chroma.from_documents(documents, embedding_model, persist_directory=None)
+    try:
+        vectorstore = Chroma.from_documents(documents, embedding_model, persist_directory=None)
+    except:
+        streamlit.error('Could not instantiate the vectorstore.')
```

Comment:
Minor - Let's log these too as well as streamlit please. 
Anywhere there's a streamllit.warning or error we should also be logging it.